### DISCUSSION TITLE
Registration Form Step 31

### DISCUSSION THREAD

**Discussion by Skye404** [USU-ITLS-x265-23-summer/class-repo#94] [Q&A] [answered]
<img width="848" alt="Screenshot 2023-06-01 202933" src="https://github.com/USU-ITLS-x265-23-summer/class-repo/assets/133051222/13d0c43c-7234-46d5-9dc1-aa943b0b7158">

I'm stuck on this step. Not sure where to go next. I don't fully understand what "id" and "type" add to the code either.

**Comment by derekscholes** [comment]
The first thing I see is the <label> opening. You need to have the for="" tag. In addition, "picture" is misspelled in your code. This may be the reason why it is not letting you finish this step. In addition, when you give a name to "id", I think it requires a dash or some sort of connector in the name as to be only one word instead of two. It should also be the same as the "for" statement in the opener. Other steps also have you name your label. If you still can't get it, let me know and I can post my code.

**Reply by derekscholes** [reply]
Oh, to answer your questions on "id" and "type". "id" connects the input to the "for" statement that needs to be in the opening label if I understood correctly. "type" helps the browser know what you want done in the label. This is why if the type="file" it allows you to upload a file. If type="checkbox" it will have a checkbox in it, etc.

**Reply by Skye404** [reply]
<img width="877" alt="Screenshot 2023-06-03 115602" src="https://github.com/USU-ITLS-x265-23-summer/class-repo/assets/133051222/7f8074c7-50c7-4b35-b518-91b1bc10c455">
Here's what I got now.

**Comment by jakegros** [comment]
It looks like you have a few extra things going on than what I needed to do.
You are going to have a label element with an input element nested inside it.  The input will include a file type.

**Reply by jakegros** [reply]
The 'for' and 'id' attributes are not needed here.

**Reply by derekscholes** [reply]
good to know. Thanks. I assumed it was since it was asked in the previous label and input examples.

**Reply by jakegros** [reply]
For sure. It was quite deceptive.

**Reply by Skye404** [reply]
I've removed the id and for attributes, but it's still not accepting it.

<img width="840" alt="Screenshot 2023-06-03 222826" src="https://github.com/USU-ITLS-x265-23-summer/class-repo/assets/133051222/72a8952f-eb93-41c4-8128-964b73f777ba">

I found this video, and I've tried it this way as well, but it's not accepting my code. https://www.youtube.com/watch?v=53E1jA-xJrA

**Reply by jakegros** [reply, marked as answer]
Frustrating.  As far as I can tell, that is what I have.
Have you reset the lesson just to clear out any junk?
![Screen Shot 2023-06-03 at 11 05 14 PM](https://github.com/USU-ITLS-x265-23-summer/class-repo/assets/133597504/22ff3791-aa25-4401-b242-df31cc10979d)

**Reply by Skye404** [reply]
<img width="871" alt="Screenshot 2023-06-05 092110" src="https://github.com/USU-ITLS-x265-23-summer/class-repo/assets/133051222/45f4118e-237d-46a2-9cd4-a0b7e201f82a">
It finally worked. I reset the lesson and reloaded the page.
